src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -379,6 +564,11 @@ ShardArchiveHandler::remove(std::lock_guard<std::mutex>&)
     app_.getShardStore()->removePreShard(shardIndex);
     archives_.erase(shardIndex);
 
+    auto db = conn_->checkoutDb();
```

Comment:
No need for a LockedSociSession. conn_ is always used while holding the lock, getSession() can be used instead.

Reply:
Done